src/network.zig

```diff
@@ -1289,6 +1289,106 @@ pub const Protocols = struct {
 			conn.send(.fast, id, writer.data.items);
 		}
 	};
+	pub const explode = struct {
+		pub const id: u8 = 14;
+		pub const asynchronous = false;
+		fn receive(conn: *Connection, reader: *utils.BinaryReader) !void {
+			const pos = try reader.readVec(Vec3i);
+
+			if(conn.user == null) return error.InvalidPacket;
+			if(!main.server.world.?.allowExplosives) return;
+			doExplode(pos);
+		}
+		fn doExplode(pos: Vec3i) void {
```

Comment:
I do not believe this belongs in the network code. Maybe the server world would be a more appropiate place?

Reply:
Well, you could just propagate it as a state change, then you can check on the server side, if the gamerule is set.
Implementing it as a state change could also allow adding a short delay and some fuse particles before detonating.